docs/source/topics/provenance/caching.rst

```diff
@@ -53,10 +57,26 @@ In order to figure out why a calculation is *not* being reused, the :meth:`~aiid
     ]
 
 
+.. _topics:provenance:caching:control:
+
+Controlling hashing
+-------------------
+
+Below are some methods you can use to control how the hashes of calculation and data classes are computed:
+
+* To ignore specific attributes, a :py:class:`~aiida.orm.nodes.Node` subclass can have a ``_hash_ignored_attributes`` attribute.
+  This is a list of attribute names, which are ignored when creating the hash.
+* For calculations, the ``_hash_ignored_inputs`` attribute lists inputs that should be ignored when creating the hash.
+* To add things which should be considered in the hash, you can override the :meth:`~aiida.orm.nodes.Node._get_objects_to_hash` method.
+  Note that doing so overrides the behavior described above, so you should make sure to use the ``super()`` method.
+* Pass a keyword argument to :meth:`~aiida.orm.nodes.Node.get_hash`.
+  These are passed on to :meth:`~aiida.common.hashing.make_hash`.
+
+
```

Comment:
This section should be clarified. Part of this is aimed at `aiida-core` developers and some of it can also be directed to plugin developers. The behavior that is described can only be adapted outside of `aiida-core` for `Data` subclasses, because process nodes cannot be subclassed. So I think this content should go in some form to the Internal Architecture section, but it should also stay here, in rewritten form, where it makes clear that this can only be done for `Data` plugins. With that given, it should also give more clear examples making it more practically targeted at plugin developers.

Reply:
In the part for plugin developers, there should also be a mention of the `invalidates_cache` keyword argument on exit codes.

Reply:
> The behavior that is described can only be adapted outside of aiida-core for Data subclasses, because process nodes cannot be subclassed.

@sphuber These are all for `Data` subclasses? Then how do you use the `_hash_ignored_inputs`? Isn't there a way to control the hashing of processes via the definition of the calcjob plugin?

> In the part for plugin developers, there should also be a mention of the `invalidates_cache` keyword argument on exit codes.

@greschd Could you explain to me how does this work? I am not familiar with it.

Reply:
> sn't there a way to control the hashing of processes via the definition of the calcjob plugin?

There are some ways to control hashing of processes, for example the `invalidates_cache` of the `ExitCode`, but anything that is defined on the `Node` class can not be changed because node classes can not be subclassed and controlled by plugins

Reply:
>  Could you explain to me how does this work? I am not familiar with it.

The `spec.exit_code` function takes an `invalidates_cache` keyword argument. If `True`, a calculation that has this exit code will not be used as a cache source.
Plugin developers can use this to e.g. disable caching for calculations that failed with an unknown error (that could well be node failure or similar). Simple example from `aiida-wannier90`: https://github.com/aiidateam/aiida-wannier90/pull/104